src/UserSpaceInstrumentation/RegisterStateTest.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+#include <stdint.h>
+#include <sys/ptrace.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+
+#include "OrbitBase/Logging.h"
+#include "UserSpaceInstrumentation/RegisterState.h"
+
+namespace orbit_user_space_instrumentation {
+
+namespace {
+
+// Let the parent trace us, write into rax and ymm0, then break. While the child is breaked the
+// parent modifies the register and continues the child. The child then reads back the registers and
+// verifies the modifications done by the parent. The exit code indicates the outcome of that
+// verification.
+void Child() {
+  CHECK(ptrace(PTRACE_TRACEME, 0, NULL, 0) != -1);
+
+  uint64_t rax = 0xaabbccdd;
+  uint8_t avx_bytes[32];
+  for (int i = 0; i < 32; ++i) {
+    avx_bytes[i] = i;
+  }
+  __asm__ __volatile__(
```

Comment:
Oh wow, please add a comment on what's happening here.

Reply:
There is a comment above the method. 
Let's take this offline and see what makes to add here.

Reply:
Added something. ptal.